### PR TITLE
fix #9981: bash complete error

### LINKED ISSUE #9981: bash auto-completion error when expanding targets in package
when expanding targets in package after typing `:` in command line:
```
$ bazel build //stage3/main:<tab>
```
auto-completion will find rules in `stage3/main/BUILD`: `hello-greet` and `hello-world`, resulting next command line as:
```
$ bazel build //stage3/main:<tab>
hello-greet   hello-world
$ bazel build hello-
```
note that the absolute path part `//stage3/main:` is erased, leaving command line in illegal state and further `<tab>` invalid, which is very annoying.

After some digging, I found that change the following line:
 
https://github.com/bazelbuild/bazel/blob/50050ae4f4f88877241ebac4f515b487fda86f67/scripts/bazel-complete-template.bash#L232

into:

```
       echo "$result" | tr " " "\n" | awk -v cur="$package_name" '{print cur":"$0;}' | sed 's|$| |'
```

would solve the problem.

I am not sure whether it is the right solution?

thanks,

### PR DESCRIPTION
**Pull request by immars**
Fixes #9981

### CODE REVIEW

**Comment by philwo** [comment]
Hi @immars! Thank you for the pull request :)

Could you check out the failing tests and see if that's something you can fix in your code?

Here's a direct link to the test log: https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/f17f8910-b79e-4f94-8838-3186991e8891/scripts/bash_completion_test/attempt_1.log

Note that it's also possible that the test is just wrong now, in case your PR improves the current behavior. In that case, could you please update the test?

**Comment by immars** [comment]
> Hi @immars! Thank you for the pull request :)
> 
> Could you check out the failing tests and see if that's something you can fix in your code?
> 
> Here's a direct link to the test log: https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/f17f8910-b79e-4f94-8838-3186991e8891/scripts/bash_completion_test/attempt_1.log
> 
> Note that it's also possible that the test is just wrong now, in case your PR improves the current behavior. In that case, could you please update the test?

Thanks for the reply,

Previously I had a look at the failed tests, but didn't make sense of them. According to the test result, my change is wrong, but in the actual bash environment my change works. 

I will try digging deeper later.

**Comment by immars** [comment]
@philwo I've fixed bash completion tests, however several other tests are failing.

Is there any guidance on how to run those tests?

Currently I have my desktop environment ubuntu 16.04, but failed to run failed tests,  `//src/test/shell/bazel:bazel_coverage_java_jdk10toolchain_released_test` for example:

```
ERROR: /home/immars/.cache/bazel/_bazel_immars/b847d7e8c6dbf366df4f7d0845f65dcc/execroot/io_bazel/_tmp/f5b36aee58d4818e59937e7dfda0160f/root/3707f93d3580ef3c0f617f1870954518/external/bazel_tools/tools/jdk/BUILD:68:1: @bazel_tools//tools/jdk:legacy_current_java_toolchain depends on @remote_java_tools_javac10_test_linux//:toolchain in repository @remote_java_tools_javac10_test_linux which failed to fetch. no such package '@remote_java_tools_javac10_test_linux//': java.io.IOException: Error downloading [https://mirror.bazel.build/bazel_java_tools/releases/javac10/v5.0/java_tools_javac10_linux-v5.0.zip] to /home/immars/.cache/bazel/_bazel_immars/b847d7e8c6dbf366df4f7d0845f65dcc/execroot/io_bazel/_tmp/f5b36aee58d4818e59937e7dfda0160f/root/3707f93d3580ef3c0f617f1870954518/external/remote_java_tools_javac10_test_linux/java_tools_javac10_linux-v5.0.zip: Checksum was d4fbed2bb22634835e42568c2d26df34fdd7281d9a7061c537f32c9970316e38 but wanted 52e03d400d978e9af6321786cdf477694c3838d7e78c2e5b926d0244670b6d3c
```

**Comment by immars** [comment]
seems that the failed test is the same reason:

https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/b286cd10-cd8d-440f-bbcc-d7df853d0984/src/test/shell/bazel/bazel_coverage_java_jdk10toolchain_released_test/attempt_3.log

**Comment by philwo** [comment]
@immars Just wanted to let you know that I'm currently on vacation, but will get back to your PR on Wednesday (2019-10-30) 😊 Thanks for working on this!

**Comment by immars** [comment]
> @immars Just wanted to let you know that I'm currently on vacation, but will get back to your PR on Wednesday (2019-10-30)  Thanks for working on this!

@philwo sure, good to know, thanks! Now I've changed the checksum in `testdata/jdk_http_archives` and the tests passed. But I'm not sure it's the right thing to do.

**Comment by immars** [comment]
@philwo  any updates? 

thanks,

**Comment by philwo** [comment]
Hi @immars,

I wanted to repro the broken behavior and then the fixed behavior before merging, but somehow I can't repro the failure on my Mac (macOS 10.14.6) with Bazel 1.1.0 (installed from the Homebrew main repository). I have attached a screen recording. Any idea?

Issue9981

**Comment by immars** [comment]
Hi,

thanks to your finding,

my bash version on mac: 4.4.23,
bash version on ubuntu: 4.3.48.

both with oh-my-bash installed. https://github.com/ohmybash/oh-my-bash

I found that some other bash-completion scripts in oh-my-bash uses:

```
export COMP_WORDBREAKS=${COMP_WORDBREAKS/\:/}
```

to deal with colons, and that interferes with bazel's bash-completion.

**Comment by meisterT** [comment]
What's the status of this PR?

**Comment by aiuto** [comment]
ping everyone

File: scripts/bash_completion_test.sh [hunk 1]
```diff
@@ -663,24 +669,24 @@ test_target_expansion_in_subdir() {
 
     # Relative labels:
     assert_expansion 'build streamer2:ta' \
-                     'build streamer2:task_lib '
+                     'build streamer2:streamer2:task_lib '
```
Comment (meisterT) [inline scripts/bash_completion_test.sh:672]: is this (here and in other places) intended to duplicate the `streamer2:` part?

**Comment by meisterT** [comment]
Closing for lack of activity, please reply if you want to merge this.

**Pull request closed without merging**